lib/Analysis/OpFoldResultUtils.cpp

```diff
@@ -91,7 +85,7 @@ OpFoldResult addOFRs(const OpFoldResult lhs, const OpFoldResult rhs,
         b.create<arith::ConstantOp>(loc, b.getIndexAttr(lhsIntAttr.value()));
     lhsValue = lhsOp.getResult();
   } else {
-    assert(isa<IndexType>(lhsValue.getType()));
+    assert(lhsValue && isa<IndexType>(lhsValue.getType()));
```

Comment:
What was the reason that this one had to change? It seems like we expect `lhsValue` to be non-null in the else case (same with `rhsValue`). I think a better change here would be to not initialize lhsValue at the declaration and instead initialize it in each branch. Like the following:

```c++
Value lhsValue;
if (...) {
  ...
  lhsValue = lhsOp.getResult();
} else {
  lhsValue = cast<Value>(lhs);
  assert(...);
}
```

Make sure to use `cast` and not `dyn_cast`. Do the same with `rhsValue`.

Reply:
The reason for this change is, Null pointer deference is Undefined behavior. The code you suggesting will behave the same, cast will crash if lhs cannot be casted to Value. But I agree it is better, updated, thank you!